numpyro/util.py

```diff
@@ -197,26 +200,28 @@ def fori_collect(lower, upper, body_fun, init_val, transform=identity,
         collected along the leading axis of `np.ndarray` objects.
     """
     assert lower <= upper
-    collection_size = upper - lower if collection_size is None else collection_size
-    assert collection_size >= upper - lower
+    assert thinning >= 0
+    collection_size = (upper - lower) // thinning if collection_size is None else collection_size // thinning
```

Comment:
@fehiepsi not quite sure what the interaction should be when `collection_size is not None`